@here/harp-mapview/lib/poi/PoiManager.ts

```diff
@@ -450,10 +394,75 @@ export class PoiManager {
             tile.tileKey.level
         );
 
-        if (textElement !== undefined) {
-            // If the poi icon is rendered, the label that shows its text should also be rendered.
-            // The distance rule of the icon should apply, not the one for text (only) labels.
-            textElement.ignoreDistance = true;
+        // If the poi icon is rendered, the label that shows text should also be rendered.
+        // The distance rule of the icon should apply, not the one for text (only) labels.
+        textElement.ignoreDistance = false;
+        tile.addTextElement(textElement);
+    }
+
+    /**
+     * Create and add POI [[TextElement]]s to tile with a series of positions.
+     */
+    private addPoi(
+        tile: Tile,
+        poiGeometry: PoiGeometry,
+        technique: PoiTechnique,
+        techniqueIdx: number,
+        positions: THREE.BufferAttribute
+    ) {
+        if (poiGeometry.stringCatalog === undefined) {
+            return;
+        }
+
+        let imageTextureName: string | undefined =
+            technique.imageTexture !== undefined
+                ? composeTechniqueTextureName(technique.imageTexture, technique)
+                : undefined;
+
+        const poiTechnique = technique as PoiTechnique;
+        const poiTableName = poiTechnique.poiTable;
+        let poiName = poiTechnique.poiName;
+
+        for (let i = 0; i < positions.count; ++i) {
+            const x = positions.getX(i);
+            const y = positions.getY(i);
+            const z = positions.getZ(i);
+
+            assert(poiGeometry.texts.length > i);
+            const text: string = poiGeometry.stringCatalog[poiGeometry.texts[i]] || "";
+            if (poiGeometry.imageTextures !== undefined && poiGeometry.imageTextures[i] >= 0) {
+                assert(poiGeometry.imageTextures.length > i);
+                imageTextureName = poiGeometry.stringCatalog[poiGeometry.imageTextures[i]];
```

Comment:
imageTextureName defined out of loop above gets overwritten and will never be restored. So if next loop iteration will fail this ```if``` condition it will get imageTextureName from previous loop iteration, and not default defined above.

If this is desired behavior, there should be some description comment. And if not, I suggest to reset this variable to default value at the beginning of each loop iteration.

Reply:
Good finding, will fix!